### ISSUE TITLE
Add flag to switch over to using CIDv1

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#3243]
To make it the switch over easier, lets get an option in as soon as possible to make go-ipfs use and generate version 1 CIDs for all internally created data. This will still be using the existing protobuf encoding for everything, the only difference will be that instead of a bare multihash (CIDv0) being used to address blocks and dags, we will use a full CID

**Comment by whyrusleeping** [comment]
@kevina this is done, yeah?

**Comment by kevina** [comment]
@whyrusleeping Depends on your definition of done.  The merged p.r., https://github.com/ipfs/go-ipfs/pull/3743, only adds CidV1 support to the add command, other commands might need it and I also think it should be a global option not one local to add.

**Comment by whyrusleeping** [comment]
@kevina fair enough, lets keep this one open.

**Comment by nitishm** [comment]
I am interested in helping with this. How can I get a list of command that might need this option ?

**Comment by kevina** [comment]
Yeah.  My current things is we should do this via a config option.  Switching to CIDV1 will break tones of stuff as we just expect a Cidv0 in many places, for example `gx`.  Then we also need to go through and fix _all_ the test cases which is fairly labor intensive, plus I am sure we will keep finding lingering bugs for months to years to come if we just switch everything to CIDv1.

The switch to `base32` will likely come first.  This switch will convert CIDv0 to CIDv1 for display, but the internal links in a dag will still be CIDv0.  This will break less as we have hacks in place already to look up CIDs by both CID versions and also CIDs can be converted back to Version 0 when required.

The switch to CIDv1 should come later.  My thinking is provide a config and default it to CIDv1.  This config option will effect most commands but not really low-level stuff so this should break less.

This still needs a lot of thought though.

**Comment by lidel** [comment]
Can be closed.

This issue asked for a flag to switch go-ipfs to generate CIDv1 instead of CIDv0. That's been available for years via `--cid-version 1` on `ipfs add` and also more recent [Import.CidVersion](https://github.com/ipfs/kubo/blob/master/docs/config.md#importcidversion) acts as a persistent config option, so user no longer needs to pass the flag every time. 

Running `ipfs config profile apply unixfs-v1-2025`
switches the node to CIDv1 with raw leaves and all the recommended ([IPIP-499](https://github.com/ipfs/specs/pull/499)) defaults in one shot.

The original request -- a way to make CIDv1 the default -- is fully addressed.
